src/PrestaShopBundle/Form/Admin/Product/ProductInformation.php

```diff
@@ -296,6 +297,27 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $event->setData($data);
             }
 
+            /*
+             * Remove duplicates to prevent SQL errors.
+             */
+            if (isset($data['features'])) {
+                $ids = [];
+                foreach ($data['features'] as $idx => $feature) {
+                    if (empty($feature['value'])) {
+                        continue;
+                    }
+
+                    $id = sprintf('%d-%d', $feature['feature'], $feature['value']);
+                    if (in_array($id, $ids)) {
+                        unset($data['features'][$idx]);
```

Comment:
isn't it dangerous to unset an element of the array while going through it?
you should either copy the array and go though the copy, or instantiate a new array and populate it while going through the initial one

Reply:
Nope it's not dangerous and because it's a data we really want to remove in this array :)
If I copy the array into another, the $_POST and other will not be changed and the legacy code will still using the wrong data.

Reply:
alright, I understand that you need to unset this array specificly
but you are sur there's not a chance you miss an index if you unset one during the loop? does php stores/saves each index? or the Iterator interface is able not to get lost?

Reply:
It always clean the current index and not the one after.
Nothing can be lost :)

Reply:
alright, good to know!! :)
all good then